tasklist/client/src/modules/components/FormJSRenderer/index.tsx

```diff
@@ -88,6 +93,70 @@ function useScrollToError(manager: FormManager) {
   );
 }
 
+function extractFilePath(data: object, path: string = ''): Map<string, string> {
+  let paths = new Map();
+
+  if (data === null || data === undefined) {
+    return paths;
+  }
+
+  for (const [key, value] of Object.entries(data)) {
+    const itemPath = path.length === 0 ? key : `${path}.${key}`;
+
+    if (typeof value === 'string' && value.startsWith('files::')) {
+      paths.set(value, itemPath);
+      continue;
+    }
+
+    if (Array.isArray(value)) {
+      const result = value
+        .map((dynamicListItem, index) =>
+          extractFilePath(dynamicListItem, `${itemPath}[${index}]`),
+        )
+        .reduce((acc, cur) => new Map([...acc, ...cur]), new Map());
+
+      paths = new Map([...paths, ...result]);
+      continue;
+    }
+
+    if (
+      typeof value === 'object' &&
+      value !== null &&
+      Object.keys(value).length > 0
+    ) {
+      const result = extractFilePath(
+        value,
+        path.length === 0 ? key : `${itemPath}.${key}`,
+      );
+      paths = new Map([...paths, ...result]);
+      continue;
+    }
+  }
+
+  return paths;
+}
+
+function injectFileMetadataIntoData(options: {
+  data: Record<string, unknown>;
+  fileMetadata: Map<string, FileUploadMetadata[]>;
+  pathsToInject: Map<string, string>;
+}): Record<string, unknown> {
+  const {data, fileMetadata, pathsToInject} = options;
+  let result = data;
+
+  pathsToInject.forEach((filepickerPath, fileKey) => {
+    const metadata = fileMetadata.get(fileKey);
+
+    if (metadata === undefined) {
+      return;
+    }
+
+    result = set(result, filepickerPath, metadata);
+  });
+
+  return result;
+}
+
```

Comment:
This is necessary because on `form-js` we can have inputs inside groups and lists which makes it hard to find the path of filepickers in the schema. This is also not trivial to do on the `form-js` so for simplicity sake we decided to keep it on Tasklist now, we might change that in the future.

We need the paths because after uploading the documents we need to replace the file key from the file store with the metadata by document upload endpoint.

### Example
We submit a form with with a filepicker nested inside a group and dynamiclist which outputs the data like this:
```json
{
  "group": {
    "list" [ 
      {
        "filepicker_input": "file::key"
      }
    ]
  }
}
```

Then after a successful upload we will enrich this data with the metadata returned by the endpoint like so:
```json
{
  "group": {
    "list" [ 
      {
        "filepicker_input": [/** metadata file0 */, /** metadata file1 */]
      }
    ]
  }
}
```